File: ClimbingStairs.py
class Solution:
    # @param n, an integer
    # @return an integer
    def climbStairs(self, n):
        if n<2:
            return 1
        a0=a1=1
        for i in range(1,n):
            (a0,a1)=(a1,a0+a1)
        return a1

n=3
print(Solution().climbStairs(n))
        

File: cpp/线性表/数组/Climbing Stairs.cpp
#include <cmath>
using namespace std;
class Solution {
  public:
    int climbStairs (int n) {
        const double s = sqrt(5);
        return floor((pow((1+s)/2,n+1) + pow((1-s)/2,n+1))/s + 0.5);
    }
};

/*
class Solution {
  public:
    int climbStairs(int n) {
        int prev = 0;
        int cur = 1;
        for(int i = 1; i <= n ; ++i){
            int tmp = cur;
            cur += prev;
            prev = tmp;
        }
        return cur;
    }
};*/